src/lightning/pytorch/loops/training_epoch_loop.py

```diff
@@ -152,10 +153,15 @@ def reset(self) -> None:
             trainer = self.trainer
             if trainer.num_training_batches != float("inf"):
                 expected_steps = math.ceil(trainer.num_training_batches / trainer.accumulate_grad_batches)
-                if self.global_step % expected_steps != 0:
+                loader = trainer.fit_loop._combined_loader
+                assert loader is not None
```

Comment:
can we have rather raise an exception?

Reply:
The combined loader will always be defined during ordinary use